src/schemathesis/schemas.py

```diff
@@ -63,24 +65,44 @@ def get_full_path(self, path: str) -> str:
     def get_all_endpoints(
         self, filter_method: Optional[Filter] = None, filter_endpoint: Optional[Filter] = None
     ) -> Generator[Endpoint, None, None]:
-        paths = self.raw_schema["paths"]  # pylint: disable=unsubscriptable-object
+        schema = deepcopy(self.raw_schema)  # modifications are going to happen
+        paths = schema["paths"]  # pylint: disable=unsubscriptable-object
         for path, methods in paths.items():
             full_path = self.get_full_path(path)
             if should_skip_endpoint(full_path, filter_endpoint):
                 continue
+            common_parameters = methods.pop("parameters", [])
             for method, definition in methods.items():
                 if should_skip_method(method, filter_method):
                     continue
                 # Maybe decompose definition into smaller parts? something is not needed probably
-                parameters = definition.get("parameters", [])
+                parameters = itertools.chain(definition.get("parameters", []), common_parameters)
+                # a parameter could be either Parameter Object or Reference Object.
+                # references should be resolved here to know where to put the parameter - body / query / etc
                 parameters = [self.prepare_item(item) for item in parameters]
                 yield Endpoint(path=full_path, method=method.upper(), parameters=parameters)
 
     def prepare_item(self, item: Dict[str, Any]) -> Dict[str, Any]:
         if is_reference(item):
-            item = self.dereference(item["$ref"])
+            item = self.resolve_reference(item["$ref"])
+        elif item["in"] == "body" and is_reference(item["schema"]):
+            # TODO. is it ok to modify it? or better to copy it?
+            item.update(self.resolve_reference(item["schema"]["$ref"]))
+            del item["schema"]
```

Comment:
Hm, it seems that I added a bug here.

If this one will be deleted, then if another test under the same `parametrize` call will need this parameter somehow (not sure if it is the case, tho), then this method will fail on the `elif` line above, because `schema` is absent in the dictionary. Maybe it is the case for common parameters, that are reused among multiple methods